aws_lambda_builders/workflows/nodejs_npm/actions.py

```diff
@@ -0,0 +1,110 @@
+"""
+Action to resolve NodeJS dependencies using NPM
+"""
+
+import logging
+from aws_lambda_builders.actions import BaseAction, Purpose, ActionFailedError
+from .npm import NpmExecutionError
+
+LOG = logging.getLogger(__name__)
+
+
+class NodejsNpmPackAction(BaseAction):
+
+    """
+    A Lambda Builder Action that packages a Node.js package using NPM to extract the source and remove test resources
+    """
+
+    NAME = 'NpmPack'
+    DESCRIPTION = "Packaging source using NPM"
+    PURPOSE = Purpose.COPY_SOURCE
+
+    def __init__(self, artifacts_dir, scratch_dir, manifest_path, osutils, subprocess_npm):
+        """
+        :type artifacts_dir: str
+        :param artifacts_dir: an existing (writable) directory where to store the output.
+            Note that the actual result will be in the 'package' subdirectory here.
+
+        :type scratch_dir: str
+        :param scratch_dir: an existing (writable) directory for temporary files
+
+        :type manifest_path: str
+        :param manifest_path: path to package.json of an NPM project with the source to pack
+
+        :type osutils: aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils
+        :param osutils: An instance of OS Utilities for file manipulation
+
+        :type subprocess_npm: aws_lambda_builders.workflows.nodejs_npm.npm.SubprocessNpm
+        :param subprocess_npm: An instance of the NPM process wrapper
+        """
+        self.artifacts_dir = artifacts_dir
+        self.manifest_path = manifest_path
+        self.scratch_dir = scratch_dir
+        self.osutils = osutils
+        self.subprocess_npm = subprocess_npm
+
+    def execute(self):
+        """
+        Runs the action.
+
+        :raises lambda_builders.actions.ActionFailedError: when NPM packaging fails
+        """
+        try:
+            package_path = "file:{}".format(self.osutils.abspath(self.osutils.dirname(self.manifest_path)))
```

Comment:
@sanathkr Will this `osutils.abspath` and `osutils.dirname` be replaced with the PathResolve changes? We are doing path things here and using `os.path` style operations which makes this harder to read. If instead we had actually objects (like pathlib), these are easier to maintain.

Or even if `osutils` returned `Path`s, you will be able to operate on them normally.

Reply:
yes exactly. they will all go away when pathresolver can give you a pathlib object back

Reply:
Awesome. Just wanted to make sure